pyrealm_build_data/splash/splash_run_time_series_parallel.py

```diff
@@ -0,0 +1,258 @@
+            sf=sf_vec[day_idx],
+            tc=tair_vec[day_idx],
+            pn=pn_vec[day_idx],
+        )
+        # Update wn
+        wn_curr = splash.wn
+
+        # Store data
+        ppfd_d[day_idx] = splash.evap.solar.ppfd_d
+        rn_d[day_idx] = splash.evap.solar.rn_d
+        rnn_d[day_idx] = splash.evap.solar.rnn_d
+        sat[day_idx] = splash.evap.sat
+        lv[day_idx] = splash.evap.lv
+        pw[day_idx] = splash.evap.pw
+        psy[day_idx] = splash.evap.psy
+        econ[day_idx] = splash.evap.econ
+        cond[day_idx] = splash.cond
+        eet[day_idx] = splash.eet
+        pet[day_idx] = splash.pet
+        aet[day_idx] = splash.aet
+        wn[day_idx] = splash.wn
+        ro[day_idx] = splash.ro
+
+    return dict(
+        coords=coords,
+        wn_spun_up=wn_spun_up,
+        ppfd_d=ppfd_d,
+        rn_d=rn_d,
+        rnn_d=rnn_d,
+        sat=sat,
+        lv=lv,
+        pw=pw,
+        psy=psy,
+        econ=econ,
+        cond=cond,
+        eet=eet,
+        pet=pet,
+        aet=aet,
+        wn=wn,
+        ro=ro,
+    )
+
+
+def run_splash_time_series(
+    input_file: Path, output_file: Path, n_cores: int = 8
+) -> None:
+    """Run a SPLASH time series for gridded data."""
+
+    start = datetime.now()
+    n_real = 0
+
+    # Import the test data file
+    input_data = xarray.load_dataset(input_file)
+
+    # Create storage for the results: evap and core solar
+    # Create storage for the results: evap and core solar
+    ppfd_d = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    rn_d = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    rnn_d = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    sat = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    lv = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    pw = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    psy = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    econ = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    cond = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    eet = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    pet = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    aet = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    wn = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    ro = xarray.full_like(input_data.tmp, fill_value=np.nan)
+    wn_spun_up = xarray.full_like(input_data.elev, fill_value=np.nan)
+
+    # Create coords tuples to pass to run_one_cell
+    coords = []
+    for lat_idx, lat in enumerate(input_data.lat):
+        for lon_idx, lon in enumerate(input_data.lon):
+            # Create the SPLASH instance
+            cell_elev = input_data["elev"][lat_idx, lon_idx]
+            coords.append(
+                (
+                    lat_idx,
+                    float(lat.data),
+                    lon_idx,
+                    float(lon.data),
+                    float(cell_elev.data),
+                )
+            )
```

Comment:
Maybe a list comprehension `coords = [(float(lat.data), float(lon.data), float(cell_elev.data)) for lat in input_data.lat for lon in input_data.lon]` would be more concise.

Reply:
That's leaving out the indices and the `elev` extraction. Its on the borderline between list comprehension and loops being simpler to comprehend, but I think this _is_ a bit cleaner:

```
coords = [
    (
        lat_idx,
        float(lat.data),
        lon_idx,
        float(lon.data),
        input_data["elev"].data[lat_idx, lon_idx],
    )
    for lat_idx, lat in enumerate(input_data.lat)
    for lon_idx, lon in enumerate(input_data.lon)
]
```

Reply:
Thanks for the response! However, if `results` is directly converted to a `pd.DataFrame` as shown in the next comment, `lat_idx` and `lon_idx` would not be necessary.

Reply:
Except that it is need to isolate `cell_elev` from `input_data['elev'] and that is not included in your list comprehension.

Reply:
I guess the other thing is that this is a rarely used setup/testing script, so doesn't need finessing to quite the same degree as main package code.